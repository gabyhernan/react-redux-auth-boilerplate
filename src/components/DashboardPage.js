import React from 'react';


const DashboardPage = () => (
  <div>
  Dashboard Page Content here
  </div>
  );



  export default DashboardPage;
