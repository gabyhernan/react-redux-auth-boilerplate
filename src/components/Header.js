import React from 'react';
import { Link } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';



export const Header = ({ startLogout}) => (
  <header className="header">
  <div className="content-container">
    <div className="header__content"> {/* to distribut header on flexbox */}
    <Link className="header__title" to="/dashboard"  >
      <h1> Boilerplate </h1>
    </Link>
  {/* activeClassName lets us provide a class that is only going to get applied
  to the link when we are on that page  */}
  <button onClick={startLogout}
          className="button button--link"
     > Logout </button>
        </div>
     </div>
   </header>
  );

const mapDispatchToProps = (dispatch) => ({
  startLogout: () =>  dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
