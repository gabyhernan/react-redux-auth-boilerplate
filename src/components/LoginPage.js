import React from 'react';
import {connect} from 'react-redux';
import { startLogin } from '../actions/auth';



 export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
   <div className="box-layout__box">
    <h1 className="box-layout__title"> Boilerplate </h1>
    <p className="box-layout__caption"> Tagline for your app </p>
  <button onClick={startLogin}
          className="button"
  > Login with Google </button>
   </div>
  </div>
);

 const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
 });

export default connect(undefined, mapDispatchToProps)(LoginPage);
