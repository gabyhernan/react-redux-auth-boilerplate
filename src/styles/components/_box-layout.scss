// Box layout

.box-layout {
  align-items: center;
  background: url('/images/bg1.jpg');
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;

}

.box-layout__box {
  background:  rgba(242, 242, 242, 0.75);
  border: 2px solid #f2f2f2;
  border-radius: 3px;
  padding: $l-size $m-size;
  text-align: center;
  width: 30rem;
}

.box-layout__title {
  background: linear-gradient(to right, #dd5e89, #f7bb97);
  background: -webkit-linear-gradient(to right, #dd5e89, #f7bb97); /* Chrome 10-25, Safari 5.1-6 */
  font-size: 5rem;
  margin: 0 0 $m-size 0;
  line-height: 1;
  letter-spacing: 2.5px;
  text-transform: lowercase;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.box-layout__caption {
  font-size: 1.7rem;
}
