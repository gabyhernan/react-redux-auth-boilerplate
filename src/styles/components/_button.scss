.button {
  border: none;
  background: $button-color;
  display: inline-block; // makes sure that the vertical margin & padding gets taken into effect
  font-size: $font-size-large;
  letter-spacing: 1.2px;
  line-height: 1;
  color: $white;
  font-weight: 300;
  padding: $s-size;
  text-decoration: none;
  &.blue {
    background: rgb(108,226,217);
  }
  &.grey {
    background: #888888;
  }
}

.button--link {
    background: transparent;
  }
