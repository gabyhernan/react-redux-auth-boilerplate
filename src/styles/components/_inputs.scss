.text-input {
  border: 1px solid #cacccd;
  font-size: $font-size-large;
  font-weight: 300;
  height: 50px;
  padding: $s-size;
  &:focus {
    outline: $blue auto 3px;
  }
}

.select {
  @extend .text-input;
}

.textarea {
  @extend .text-input;
  height: 10rem;
}
