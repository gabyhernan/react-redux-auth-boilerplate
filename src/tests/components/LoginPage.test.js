import React from 'react';
import { shallow } from 'enzyme';
import {LoginPage} from '../../components/LoginPage';


test('should render Header correctly' , () => {
  const wrapper = shallow(<LoginPage />); // we pass in JSX trying to render
  expect(wrapper).toMatchSnapshot();

// better documented example in Header.test.js
});


test('should call startLogin on button click', () => {
  const startLoginSpy = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLoginSpy} />);

  wrapper.find('button').simulate('click');
  expect(startLoginSpy).toHaveBeenCalled();

});
