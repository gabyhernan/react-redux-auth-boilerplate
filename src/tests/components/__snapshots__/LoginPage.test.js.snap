// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
       Boilerplate 
    </h1>
    <p
      className="box-layout__caption"
    >
       Tagline for your app 
    </p>
    <button
      className="button"
    >
       Login with Google 
    </button>
  </div>
</div>
`;
